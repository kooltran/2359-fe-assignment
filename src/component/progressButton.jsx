import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { updateProgressBarAction } from "../redux/actions/updateProgressBarAction";

export class ProgressButton extends React.Component {
  static propTypes = {
    selectedProgress: PropTypes.number.isRequired,
    bars: PropTypes.array.isRequired,
    buttons: PropTypes.array.isRequired,
    updateProgressBarAction: PropTypes.func.isRequired
  };

  handleUpdateProgressBar = value => {
    const { bars, selectedProgress } = this.props;
    const selectedBarValue = bars[selectedProgress - 1];
    this.props.updateProgressBarAction(
      value,
      selectedProgress,
      selectedBarValue
    );
  };

  render() {
    const { buttons } = this.props;
    return (
      <div>
        {buttons.map((button, idx) => (
          <button
            className="progress-button"
            key={idx}
            onClick={() => this.handleUpdateProgressBar(button)}
          >
            {button}
          </button>
        ))}
      </div>
    );
  }
}

const mapDispatchToProps = {
  updateProgressBarAction
};

const mapStateToProps = state => {
  return {
    bars: state.barsInfoReducer.data.bars,
    selectedProgress: state.selectBarReducer.selectedProgress
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ProgressButton);
