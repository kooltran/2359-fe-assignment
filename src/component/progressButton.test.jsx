import React from "react";
import { shallow } from "enzyme";
import { ProgressButton } from "./progressButton";

describe("renders ProgressButton component", () => {
  const baseProps = {
    bars: [],
    selectedProgress: 1,
    buttons: [10, 20, 30],
    updateProgressBarAction: jest.fn()
  };

  it("should render properly with base props", () => {
    const wrapper = shallow(<ProgressButton {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("updateProgressBarAction should be call bar when click on the button", () => {
    const wrapper = shallow(<ProgressButton {...baseProps} />);
    wrapper
      .find(".progress-button")
      .at(0)
      .simulate("click");
    expect(baseProps.updateProgressBarAction).toHaveBeenCalled();
  });
});
