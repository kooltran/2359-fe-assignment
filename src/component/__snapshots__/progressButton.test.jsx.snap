// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProgressButton component should render properly with base props 1`] = `
<div>
  <button
    className="progress-button"
    key="0"
    onClick={[Function]}
  >
    10
  </button>
  <button
    className="progress-button"
    key="1"
    onClick={[Function]}
  >
    20
  </button>
  <button
    className="progress-button"
    key="2"
    onClick={[Function]}
  >
    30
  </button>
</div>
`;
